dev/src/index.ts

```diff
@@ -920,8 +920,21 @@ export class Firestore implements firestore.Firestore {
    * the transaction.
    *
    * You can use the transaction object passed to 'updateFunction' to read and
-   * modify Firestore documents under lock. Transactions are committed once
-   * 'updateFunction' resolves and attempted up to five times on failure.
+   * modify Firestore documents under lock. You have to perform all reads before
+   * before you perform any write.
+   *
+   * Documents read during a transaction are locked pessimistically. A
+   * transaction's lock on a document blocks other transactions, batched
+   * writes, and other non-transactional writes from changing that document.
+   * A transaction releases its document locks at commit time or once it times
+   * out or fails for any reason.
+   *
+   * Transactions are committed once 'updateFunction' resolves. If a transaction
+   * fails with contention, the transaction is retried five up to five times and
+   * its `updateFunction` is invoked again.
+   *
+   * Transaction time out after 270 seconds or 60 seconds of after the last
```

Comment:
nits:
s/Transaction/Transactions
s/60 seconds of after/seconds after
"last document is read"

Also, when does the timeout change from 60s to 270s? If a document is read in a tx, is the timeout window reduced to 60s + write-only tx timeout have a 270s timeout?

Reply:
Changed to:

```
   * Transactions time out after 60 seconds if no documents are read.
   * Transactions that are not committed within than 270 seconds are also
   * aborted.
 ```